GitUI/CommandsDialogs/FormRemotes.cs

```diff
@@ -365,14 +373,20 @@ private void SaveClick(object sender, EventArgs e)
                         FireRemoteRenamedEvent(new RemoteRenamedEventArgs(_selectedRemote.Name, remote));
                     }
 
-                    var remotes = Repositories.RemoteRepositoryHistory.Repositories;
-                    RemoteUpdate(remotes, _selectedRemote?.Url, remoteUrl);
-                    if (checkBoxSepPushUrl.Checked)
+                    ThreadHelper.JoinableTaskFactory.Run(async () =>
                     {
-                        RemoteUpdate(remotes, _selectedRemote?.PushUrl, remotePushUrl);
-                    }
-
-                    Repositories.SaveSettings();
+                        var repositoryHistory = await RepositoryHistoryManager.Remotes.LoadHistoryAsync();
+
+                        await this.SwitchToMainThreadAsync();
+                        RemoteUpdate(repositoryHistory, _selectedRemote?.Url, remoteUrl);
+                        if (checkBoxSepPushUrl.Checked)
+                        {
+                            RemoteUpdate(repositoryHistory, _selectedRemote?.PushUrl, remotePushUrl);
+                        }
+
+                        await TaskScheduler.Default;
```

Comment:
:bulb: Likely unnecessary context switch (assuming `SaveHistoryAsync` is truly asynchronous)